pkg/ebpf/tracer.go

```diff
@@ -369,6 +387,9 @@ func (t *Tracer) getConnections(active []ConnectionStats) ([]ConnectionStats, ui
 
 	t.conntracker.ClearShortLived()
 
+	// Resolve IPs to names
+	active = t.resolveConnections(active...)
```

Comment:
It'd be simpler to just do `t.reverseDNS.Resolve(connections)` and have it update the connections in place. It's kind of what's happening already

Reply:
Ah I see it returns `[]NamePair`.  I thought we were operating with `[]*ConnectionStats` everywhere. Is that not the case?

Reply:
no, we actually use `[]ConnectionStats` everywhere (prior to the protobuf serialization), but that still enables us to update in-place the structs, if you think that's a better approach.

TBH initially I thought of method signature that was entirely decoupled of the idea of `ConnectionStats` like `Resolve([]util.Address)[]string`. But that would be too cumbersome for callers and it would result in a bunch of unnecessary allocations. But in the end I find this `[]NamePair` result awkward too. Happy to change it.

Reply:
As per our discussion IRL I won't be changing this in this PR. A follow-up PR will remove the DNS fields from `ConnectionStats` altogether and we'll have DNS mappings at the payload level.